transform-plugins/src/main/java/io/cdap/plugin/XMLParser.java

```diff
@@ -77,66 +82,79 @@ public XMLParser(Config config) {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     super.configurePipeline(pipelineConfigurer);
-    outSchema = config.getOutputSchema();
+    // Get failure collector for updated validation API
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    outSchema = config.getOutputSchema(collector);
+    if (!this.config.containsMacro(INPUT)
+        && pipelineConfigurer.getStageConfigurer().getInputSchema().getField(this.config.inputField) == null) {
```

Comment:
this is difficult to read, it's better to have a separate variable for the input schema.

Also, the input schema can be null. This check should be skipped if it's null.

Reply:
Fixed